transport/src/commonTest/kotlin/com/genesys/cloud/messenger/transport/network/test_engines/RetryEngine.kt

```diff
@@ -0,0 +1,70 @@
+package com.genesys.cloud.messenger.transport.network.test_engines
+
+import com.genesys.cloud.messenger.transport.respondNotFound
+import io.ktor.client.HttpClientConfig
+import io.ktor.client.engine.mock.MockEngineConfig
+import io.ktor.client.engine.mock.respond
+import io.ktor.http.HttpStatusCode
+
+/**
+ * Test engine for 5xx server error retry scenarios
+ */
+internal fun HttpClientConfig<MockEngineConfig>.retryEngine() {
+    engine {
+        var requestCount = 0
+
+        addHandler { request ->
+            requestCount++
+
+            when (request.url.toString()) {
+                "https://test.com/server-error-502" -> {
+                    if (requestCount <= 3) {
+                        respond(
+                            status = HttpStatusCode.BadGateway,
+                            content = "Bad Gateway"
+                        )
+                    } else {
+                        respond("Success after server error retry")
+                    }
+                }
+
+                "https://test.com/server-error-503" -> {
```

Comment:
You do not need to configure different responses per every failed HttpStatusCode, unless you actually use them for testing. In this pr we test only `https://test.com/server-error-502` so the rest of the branches are not used.